You can now print pending, approved, and fulfilled orders [Ready]

WhiteHusky opened the pull request:
- [X] Can you print the receipts?
- [ ] ~~Did the bugfix actually fix an existing problem?~~
  - There's no hidden contraband supplypacks, yet, but at least this will make sure it'll regenerate if the emag state changes.
- [X] Is it stable?
  - ~~Something did break. For some reason after opening and closing a few contents it causes the browser to freeze and crash Byond with it. Variables seem sane and attempting to use the computer/program in that current state will crash the client again until it's closed (force restart of the computer). [See here.](https://puu.sh/CkTna.webm)~~ Feature removed. See bottom.

Todo:
- [X] Give more room for buttons
  - Decreased sized of other columns.
- [ ] ~~Maybe redesign the order view?~~
  - No longer needed.

![image](https://user-images.githubusercontent.com/3196371/50319662-7bcabb00-0496-11e9-93e6-b2b681ce69dc.png)
~~[Show Contents Feature](https://user-images.githubusercontent.com/3196371/50319780-0f9c8700-0497-11e9-9bcf-27410ea3b49c.png)
Added the ability to see contents. **This is broken and causes the browser to crash randomly. Unsure why.**~~
Feature removed and will be part of a different PR. The other feature of printing receipts works fine and there's no good reason to halt that for much longer because of it's own issue.
:cl:
rscadd: You can now print receipts on pending, approved, and archived orders.
/:cl:

comma commented:
>Can you print the receipts?
As long as you can't combine them with crafted crates for free cargo points

WhiteHusky commented:
> > Can you print the receipts?
> > As long as you can't combine them with crafted crates for free cargo points

I believe the piece of paper that comes with orders is tagged with the order in question, but it's been awhile since I've looked into that code. I'll get a copy of source on my laptop and look into it.

HeyBanditoz commented:
It's the stamped manifest that earns cargo points, not the receipt.

WhiteHusky commented:
Back at my main rig. Updated the OP with additional information, particularly about a strange bug that's causing the browser to crash and bring Byond down with it.
![Video](https://puu.sh/CkTna.webm)

afterthought2 commented on code/modules/modular_computers/file_system/programs/generic/supply.dm:
```diff
@@ -90,20 +103,39 @@
+/datum/nano_module/supply/proc/find_order_by_id(var/order_id, var/list/find_in)
+	for(var/datum/supply_order/SO in find_in)
+		if(SO.ordernum == order_id)
+			return SO
```
Everything below this line in this proc is unnecessary.

WhiteHusky replied:
Was trying to be verbose, but applied the suggestion.

afterthought2 commented on code/modules/modular_computers/file_system/programs/generic/supply.dm:
```diff
@@ -173,41 +205,76 @@
-				SSsupply.shoppinglist -= SO
-				SSsupply.points += SO.object.cost
-				break
+		var/datum/supply_order/SO = find_order_by_id(order_id = id, find_in = SSsupply.shoppinglist)
```
Using named arguments unnecessarily is discouraged, as it suggests to the reader that these arguments are out of position (it can also break certain things, or make the more fragile, though that's not relevant here).

WhiteHusky replied:
_Ditto_

afterthought2 reviewed (commented):
I'm not sure why it's crashing. Check if it's crashing the server: if so, find which href crashes it and look for loops, etc. If not, try removing the template macro definitions and see if that helps.

WhiteHusky commented:
@afterthought2 Doesn't seem to be crashing the server in the process. The server runs dandily when the client is crashing, but I'll remove the templates and see if it sorts out the issue.

WhiteHusky commented:
@afterthought2 After attending things, I got around to testing if it crashes without the templating and unfortunately and fortunately it still crashes but at least it isn't because of the template.

WhiteHusky committed:
Resolved conflicts with dev

Reverts the show contents feature.

That will be part of a different PR to more closely figure out why it's crashing.

Cleanup and minor optimizations.

You can now print pending, approved, and fulfilled orders.

WhiteHusky commented:
Updated OP with new information. Show contents feature removed for to be in a different PR to figure out why it's breaking.

WhiteHusky commented:
Fixed the crashing issue and resolved conflicts with dev. Turns out it was variable collision with the for loop verses the nested on. Nested for loop uses different variables and thus is now fixed.

WhiteHusky commented:
Resolved conflicts.
